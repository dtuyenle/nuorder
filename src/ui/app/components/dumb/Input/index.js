import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';
import thm from '../../../helpers/theme';

/**
 * Component`Input`.
 */
const Input = styled.input`
  width: 100%;
  border: 4px solid ${thm.borderColor};
  border-radius: ${props => props.borderRadius};
  font-size: ${thm.fontSizes.delta};
  padding: ${props => props.padding};
  margin: ${props => props.margin};

  :focus {
    outline: none;
  }
`;

Input.propTypes = {
  /** Custom margin style to be override the default. */
  margin: PropTypes.string,
  /** Custom padding style to be override the default. */
  padding: PropTypes.string,
  /** Custom border-radius style to be override the default. */
  borderRadius: PropTypes.string,
};

Input.defaultProps = {
  margin: thm.sizeUnits(0),
  padding: `${thm.sizeUnits(2)}`,
  borderRadius: '2px 0px 0px 2px',
};

/** @component */
export default Input;
