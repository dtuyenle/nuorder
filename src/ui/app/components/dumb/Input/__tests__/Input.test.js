import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';
import Input from '..';
import thm from '../../../../helpers/theme';

describe('Input component', () => {
  it('should render a default Input component', () => {
    const component = renderer.create(
      <Input />,
    ).toJSON();
    expect(component).toMatchSnapshot();
  });

  it('should have a 4px solid border and the appropriate color', () => {
    const component = renderer.create(
      <Input />,
    ).toJSON();

    expect(component).toHaveStyleRule('border', `4px solid ${thm.borderColor}`);
  });
});
