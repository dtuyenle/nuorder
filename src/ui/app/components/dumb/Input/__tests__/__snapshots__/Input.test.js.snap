// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input component should render a default Input component 1`] = `
.c0 {
  width: 100%;
  border: 4px solid #B2CCDB;
  border-radius: 2px 0px 0px 2px;
  font-size: 20px;
  padding: 16px;
  margin: 0px;
}

.c0:focus {
  outline: none;
}

<input
  className="c0"
/>
`;
